nodebuilder/settings.go

```diff
@@ -0,0 +1,27 @@
+package nodebuilder
+
+import (
+	"go.uber.org/fx"
+
+	headermodule "github.com/celestiaorg/celestia-node/nodebuilder/header"
+	"github.com/celestiaorg/celestia-node/params"
+)
+
+// WithNetwork specifies the Network to which the Node should connect to.
+// WARNING: Use this option with caution and never run the Node with different networks over the same persisted Store.
+func WithNetwork(net params.Network) fx.Option {
+	return fx.Replace(net)
+}
```

Comment:
Any actionables here or just reminiscing? :)

Reply:
No actionable. Just reminded of this discussion
![image](https://user-images.githubusercontent.com/22449364/191075738-40574084-dde1-43ed-a301-b62f1d8c209a.png)

Still don't see the logic behind the pushback there. This option should have been used from the beginning without changing the params. Moreover, it wouldn't be conflicting with this PR, because the Network option is preserved. 

So reminiscing ;)

Reply:
I dont think it belonged there, I think that it fits into its own PR - if refactorings would have already been merged by then it would be a different story